$OpenBSD: patch-src_bin_e_start_main_c,v 1.1 2012/11/10 15:06:03 stsp Exp $
- avoid Linux-specific ptrace(2) usage which doesn't compile
- instead actually launch enlightenment via execv()
--- src/bin/e_start_main.c.orig	Sat Nov  3 22:20:25 2012
+++ src/bin/e_start_main.c	Thu Nov  8 02:55:31 2012
@@ -18,7 +18,9 @@
 
 #include <Eina.h>
 
+#ifdef __linux__
 static Eina_Bool tainted = EINA_FALSE;
+#endif
 
 static void env_set(const char *var, const char *val);
 EAPI int    prefix_determine(char *argv0);
@@ -226,7 +228,9 @@ main(int argc, char **argv)
    char valgrind_path[PATH_MAX] = "";
    const char *valgrind_log = NULL;
    Eina_Bool really_know = EINA_FALSE;
+#ifdef __linux__
    Eina_Bool restart = EINA_TRUE;
+#endif
 
    eina_init();
 
@@ -376,11 +380,12 @@ main(int argc, char **argv)
    args[i++] = buf;
    copy_args(args + i, argv + 1, argc - 1);
    args[i + argc - 1] = NULL;
-   /* execv(args[0], args); */
+   execv(args[0], args);
 
    /* not run at the moment !! */
 
 
+#ifdef __linux__
    /* Now looping until */
    while (restart)
      {
@@ -497,6 +502,7 @@ main(int argc, char **argv)
           }
           
      }
+#endif /* __linux__ */
 
    return -1;
 }
